eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Gateway/DoctrineDatabase.php

```diff
@@ -97,6 +107,8 @@ public function index(FullTextData $fullTextData)
         $wordCount = 0;
         $placement = 0;
 
+        // Remove previously indexed content if exists to avoid keeping in index removed field values
+        $this->remove($fullTextData->id);
```

Comment:
Note: Till now, `WordIndexer` implementation added new words but didn't care about words that got removed from content object. Removing Content Object before reindexing it is the simplest way to fix this. Test that fails without it is [here](https://github.com/alongosz/ezpublish-kernel/blob/fbd7f92de5c4100aff35a891e1aac920a3a32759/eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php#L699).